protocol/contracts/libraries/LibDibbler.sol

```diff
@@ -31,31 +39,136 @@ library LibDibbler {
         return pods;
     }
 
-    function sowPlot(
-        address account,
-        uint256 beans,
-        uint256 pods
-    ) private {
+    /// @dev function returns the weather scaled down
+    /// @notice based on the block delta
+    // precision level 1e6, as soil has 1e6 precision (1% = 1e6)
+    // the formula log2(A * BLOCK_ELAPSED_MAX + 1) is applied, where
+    // A = 2;
+    // MAX_BLOCK_ELAPSED = 25;
+    
+    function morningAuction() internal view returns (uint256) {
         AppStorage storage s = LibAppStorage.diamondStorage();
-        s.a[account].field.plots[s.f.pods] = pods;
-        emit Sow(account, s.f.pods, beans, pods);
+        uint256 delta = block.number.sub(s.season.sunriseBlock);
+        if (delta > 24) { // check most likely case first
+            return uint256(s.w.yield).mul(DECIMALS);
+        } else if (delta == 1) {
+            return auctionMath(279415312704); 
+        } else if (delta == 2) {
+            return auctionMath(409336034395);
+        } else if (delta == 3) {
+            return auctionMath(494912626048); 
+        } else if (delta == 4) {
+            return auctionMath(558830625409);
+        } else if (delta == 5) {
+            return auctionMath(609868162219);
+        } else if (delta == 6) {
+            return auctionMath(652355825780);
+        } else if (delta == 7) {
+            return auctionMath(688751347100); 
+        } else if (delta == 8) {
+            return auctionMath(720584687295);
+        } else if (delta == 9) {
+            return auctionMath(748873234524);
+        } else if (delta == 10) {
+            return auctionMath(774327938752);
+        } else if (delta == 11) {
+            return auctionMath(797465225780); 
+        } else if (delta == 12) {
+            return auctionMath(818672068791);
+        } else if (delta == 13) {
+            return auctionMath(838245938114);
+        } else if (delta == 14) {
+            return auctionMath(856420437864);
+        } else if (delta == 15) {
+            return auctionMath(873382373802); 
+        } else if (delta == 16) {
+            return auctionMath(889283474924);
+        } else if (delta == 17) {
+            return auctionMath(904248660443);
+        } else if (delta == 18) {
+            return auctionMath(918382006208);
+        } else if (delta == 19) {
+            return auctionMath(931771138485);
+        } else if (delta == 20) {
+            return auctionMath(944490527707);
+        } else if (delta == 21) {
+            return auctionMath(956603996980);
+        } else if (delta == 22) {
+            return auctionMath(968166659804);
+        } else if (delta == 23) {
+            return auctionMath(979226436102);
+        } else if (delta == 24) {
+            return auctionMath(989825252096);
+        } else {
+            /// @dev should only occur when delta == 0 
+            /// (i.e called in the same block as sunrise) 
+            return DECIMALS; //minimium 1% yield
+        }
```

Comment:
Could reorg this long if ladder into a binary tree of if statements to improve efficiency. (Eg. First, Check if > 16. If > 16, secondly check if > 24, etc...). Would use a max of log(24) if statements instead of a max of 24 if statements. Not sure how much it improves gas on average, but could be worth investigating.

Reply:
found that the binary tree does save significant gas per sow on average (500), will refactor the function to use a binary tree.